students/hellosputnik/main.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+)
+
+type Problem struct {
+	question string
+	answer   string
+}
+
+type Quiz struct {
+	problems []Problem
+	score    int
+}
+
+type Settings struct {
+	filename  *string
+	timeLimit *int
+}
+
+func main() {
+	quiz := Quiz{}
+	settings := Settings{}
+
+	settings.filename = flag.String("csv", "problems.csv", "a csv file in the format of 'question,answer'")
+	settings.timeLimit = flag.Int("limit", 30, "the time limit for the quiz in seconds")
+
+	// Get the flags (if any).
+	flag.Parse()
+
+	// Create a file handle for the file.
+	file, err := os.Open(*settings.filename)
+
+	// If there was an error opening the file, exit.
+	if err != nil {
+		panic(err)
+	}
+
+	defer file.Close()
+
+	// Create a buffered reader to read the file input.
+	fin := bufio.NewScanner(file)
+
+	// Read the problems from the comma-separated values file.
+	for fin.Scan() {
+		line := strings.Split(fin.Text(), ",")
+		problem := Problem{question: line[0], answer: line[1]}
+
+		quiz.problems = append(quiz.problems, problem)
+	}
+
+	// Create a timer to enforce a time limit.
+	timer := time.NewTimer(time.Second * time.Duration(*settings.timeLimit))
+	defer timer.Stop()
+
+	go func() {
+		<-timer.C
```

Comment:
This will run as soon as the time limit is up, but it won't terminate your program. I cover a few ways to do this in the videos that go w/ this exercise, but one way to quickly add it to your code would be to add something like below to the end of this goroutine closure:

```go
os.Exit(0)
```

`os.Exit` will cause your entire application to exit and return a status code. You can read more about it here - https://golang.org/pkg/os/#Exit